src/main/java/io/openliberty/tools/intellij/util/LibertyMavenUtil.java

```diff
@@ -191,68 +191,103 @@ private static boolean containerVersion(String version){
+     * @throws IOException
+     */
+    private static String getLocalMavenWrapper(Project project) throws LibertyException {
         String mvnw = SystemInfo.isWindows ? ".\\mvnw.cmd" : "./mvnw";
         File file = new File(project.getBasePath(), mvnw);
-        return file.exists() ? mvnw : null;
+        if (!file.exists()){
+            String translatedMessage = LocalizedResourceUtil.getMessage("maven.wrapper.does.not.exist");
+            throw new LibertyException("A Maven wrapper for the project could not be found. Make sure to configure a " +
+                    "valid Maven wrapper or change the build preferences for Maven inside IntelliJ Maven preferences.", translatedMessage);
+        }
+        if (!file.canExecute()) {
+            String translatedMessage = LocalizedResourceUtil.getMessage("maven.wrapper.cannot.execute");
+            throw new LibertyException("Could not execute Maven wrapper because the process does not have permission to " +
+                    "execute it. Consider giving executable permission for the Maven wrapper file or changing the build " +
+                    "preferences for Maven inside IntelliJ Maven preferences.", translatedMessage);
+        }
+        return mvnw;
     }
 
-    private static String getCustomMavenPath(Project project, String customMavenHome) {
+    /**
+     * Get the custom Maven path from Built Tools in IntelliJ
+     *
+     * @param project liberty project
+     * @param customMavenHome the custom Maven Home
+     * @return Maven path to be executed or an exception to display
+     * @throws IOException
+     */
+    private static String getCustomMavenPath(Project project, String customMavenHome) throws LibertyException {
         File mavenHomeFile = MavenServerManager.getMavenHomeFile(customMavenHome); // when customMavenHome path is invalid it returns null
-        if (mavenHomeFile != null) {
-            // When a custom maven is specified, IntelliJ settings force it to point to the root folder and consider the subfolders invalid,
-            // and consequently, it will return null. For this reason, we need to use ./bin/mvn in order to execute maven.
-            File mavenExecutable = new File(mavenHomeFile.getAbsolutePath(), "bin" + File.separator + "mvn");
+        if (mavenHomeFile == null) {
+            String translatedMessage = LocalizedResourceUtil.getMessage("maven.invalid.build.preference");
+            throw new LibertyException("Make sure to configure a valid path for Maven home path inside IntelliJ Maven preferences.", translatedMessage);
+        }
+        // When a custom maven is specified, IntelliJ settings force it to point to the root folder and consider the subfolders invalid,
+        // and consequently, it will return null. For this reason, we need to use ./bin/mvn in order to execute maven.
+        File mavenExecutable = new File(mavenHomeFile.getAbsolutePath(), "bin" + File.separator + "mvn");
```

Comment:
On line 262 it says "new File (new File(mavenJdk, "bin"), "java");". Consider using that technique here and also for gradle if there is other rework you need to do.

Reply:
Also, good catch!